consensus/state.go

```diff
@@ -411,6 +415,25 @@ func (cs *State) loadWalFile() error {
 
 // OnStop implements service.Service.
 func (cs *State) OnStop() {
+
+	// If the node is committing the new block, wait until it finished!
+	if cs.GetRoundState().Step == cstypes.RoundStepCommit {
+	wait:
+		for {
+			select {
+			case rs := <-cs.onStopCh:
+				if rs != nil && rs.Step == cstypes.RoundStepNewHeight && rs.Round == 0 {
+					break wait
+				}
+			case <-time.After(time.Duration(cs.config.TimeoutCommit.Seconds())):
```

Comment:
Is it possible to just do it like this
```suggestion
			case <-time.After(cs.config.TimeoutCommit):
```